src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java

```diff
@@ -1234,13 +1235,15 @@ private static boolean labelSelectorMethodChanged(Monitor original, MonitorCU ne
 
     // If a new one is to be bound, bind it
     if (StringUtils.isNotBlank(updatedResourceId)) {
-      ResourceDTO resource  = resourceApi.getByResourceId(monitor.getTenantId(), updatedResourceId);
-      affectedEnvoys.addAll(
-          upsertBindingToResource(
-              Collections.singletonList(monitor),
-              resource,
-              null
-          ));
+      ResourceDTO resource = resourceApi.getByResourceId(monitor.getTenantId(), updatedResourceId);
+      if (!ObjectUtils.isEmpty(resource)) {
```

Comment:
Just this would work, right?
```suggestion
      if (resource != null) {
```

Reply:
Yes This would work too